pegasus/helpers/section_api_helpers.rb

```diff
@@ -526,38 +526,73 @@ def self.fields
 end
 
 class DashboardUserScript
+  # Assigns a script to all users enrolled in the section, creating a new user_scripts object if
+  # necessary. The method noops for those user_scripts that already exist with assigned_at set.
+  # WARNING: This method does not verify that the section and student_users exist (aren't deleted).
   def self.assign_script_to_section(script_id, section_id)
-    # create userscripts for users that don't have one yet
-    Dashboard.db[:user_scripts].
-      insert_ignore.
-      import(
-        [:user_id, :script_id],
-        Dashboard.db[:followers].
-          select(:student_user_id, script_id.to_s).
-          where(section_id: section_id, deleted_at: nil)
-      )
+    student_user_ids = Dashboard.db[:followers].
+      select(:student_user_id).
+      where(section_id: section_id, deleted_at: nil).
+      map {|f| f[:student_user_id]}
+    DashboardUserScript.assign_script_to_users(script_id, student_user_ids)
   end
 
+  # Assigns a script to the user via user_scripts, creating a new user_scripts object if necessary.
+  # The method noops if a user_scripts already exists with assigned_at set.
+  # @param script_id [Integer] The dashboard ID of the script.
+  # @param user_id [Integer] The dashboard ID of the user.
   def self.assign_script_to_user(script_id, user_id)
-    # creates a userscript for a user if they don't have it yet
-    Dashboard.db[:user_scripts].
-      insert_ignore.
-      import(
```

Comment:
My understanding of Sinatra Sequel is that there is no reason to use `import` for a single record insert. And, now that we are verifying the `user_scripts` doesn't already exist, we can `insert` rather than `insert_ignore`.

There is a race condition here, that I'm happily ignoring.